### PR TITLE
dojson: BAI support

### PR DESCRIPTION
**Pull request by kaplun**
* Maps 100/700__w to inspire_bai field.

* Makes inspire_bai field searchable.

* Updates demo records to have BAIs.

* Amends hep.json to support multiple editors in references.

Signed-off-by: Samuele Kaplun <samuele.kaplun@cern.ch>

### CODE REVIEW

**Comment by kaplun** [comment]
cc: @jacenkow

File: inspirehep/modules/records/jsonschemas/records/hep.json [hunk 1]
```diff
@@ -454,7 +454,10 @@
                         "type": "array"
                     },
                     "editors": {
-                        "type": "string"
+                        "type": "array",
+                        "items": {
+                            "type": "string"
+                        }
```
Comment (jacquerie) [inline inspirehep/modules/records/jsonschemas/records/hep.json]: Looks unrelated?

**Comment by jacquerie** [comment]
The change to add `inspire_bai` looks correct (haven't tested it though), but I can't understand how are the changes to the citations data related...

**Comment by kaplun** [comment]
@jacquerie they are not. But since I have refreshed the demo records, one of the record is now sporting a reference with more than one editors and I had to fix it, otherwise I wouldn't be able to load the records at all...

File: inspirehep/modules/records/jsonschemas/records/hep.json (hunk 1 above)
Comment (kaplun) [inline inspirehep/modules/records/jsonschemas/records/hep.json]: I separated it into a dedicated commit
